doc/design/gan_api.md

```diff
@@ -0,0 +1,193 @@
+# Design for GAN
+
+GAN (General Adversarial Net) is an important model for unsupervised learning and widely used in many areas. 
+
+It contains several important machine learning concepts, including building and running subgraphs, dependency tracing, different optimizers in one executor and so forth.
```

Comment:
I think `contain` is not good, maybe `It will use ...`?

Reply:
Thanks. Done.